core/include/vecmem/memory/impl/device_atomic_ref.ipp

```diff
@@ -14,41 +14,51 @@
 
 /// Helpers for explicit calls to the SYCL atomic functions
 #if defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)
-#define __VECMEM_SYCL_ATOMIC_CALL0(FNAME, PTR) \
-    cl::sycl::atomic_##FNAME<value_type>(      \
-        cl::sycl::atomic<value_type>(cl::sycl::global_ptr<value_type>(PTR)))
-#define __VECMEM_SYCL_ATOMIC_CALL1(FNAME, PTR, ARG1)                         \
-    cl::sycl::atomic_##FNAME<value_type>(                                    \
-        cl::sycl::atomic<value_type>(cl::sycl::global_ptr<value_type>(PTR)), \
+#define __VECMEM_SYCL_ATOMIC_CALL0(FNAME, PTR)           \
+    cl::sycl::atomic_##FNAME<value_type, address>( \
+        cl::sycl::atomic<value_type, address == address_space::global
+                           ? ::sycl::memory_scope::device
+                           : ::sycl::memory_scope::work_group>(     \
+            cl::sycl::global_ptr<value_type>(PTR)))
```

Comment:
I believe you'll also need to abstract this call. It shouldn't always be `sycl::global_ptr`.